### PR TITLE
arcs_kt_plan (and arcs_kt_gen) now has manifest import support

### PR DESCRIPTION
**Pull request by alxrsngrtn**
- Import support
- Small refactoring here and there
- Now the main codegen rules have a `data` attribute. This way, `deps` can be just jars, and we can specify .arcs files needed at runtime. 
- Fixed bug with create handles in the goldens

### CODE REVIEW

File: java/arcs/core/data/testdata/BUILD [hunk 1]
```diff
@@ -30,3 +31,18 @@ arcs_kt_gen(
...
     srcs = ["IngestProcessRetrieve.arcs"],
 )
+
+arcs_manifest(
+    name = "shared_schemas",
+    srcs = ["Schemas.arcs"],
+)
+
+arcs_kt_gen(
+    name = "etl_gen",
```
Comment (csilvestrini) [inline java/arcs/core/data/testdata/BUILD:41]: What is "etl"?

File: javatests/arcs/android/e2e/ArcsTests.kt [hunk 2]
```diff
@@ -96,6 +96,7 @@ class ArcsTest {
     }
 
     @Test
+    @Ignore("Broken: b/154181519")
```
Comment (csilvestrini) [inline javatests/arcs/android/e2e/ArcsTests.kt:99]: This bug is marked fixed. Is it not actually fixed? Can you reopen it with more detail?

File: src/tools/recipe2plan.ts [hunk 3]
```diff
@@ -22,13 +22,14 @@ import {assert} from '../platform/assert-node.js';
  */
 export async function recipe2plan(path: string): Promise<string> {
   return await Flags.withDefaultReferenceMode(async () => {
-    const manifest = await Runtime.parseFile(path);
+    const manifest = await Runtime.parse(`import '${path}'`);
```
Comment (csilvestrini) [inline src/tools/recipe2plan.ts:25]: Why this change?

File: src/tools/storage-key-recipe-resolver.ts [hunk 4]
```diff
@@ -128,25 +127,24 @@ export class StorageKeyRecipeResolver {
...
-
-        const match = matches[0];
-        if (!isLongRunning(match.recipe)) {
-          throw new StorageKeyRecipeResolverError(
-            `Handle ${handle.localName} mapped to ephemeral handle '${match.id}'.`
-          );
-        }
-      });
+    const isMapOrCopyFate = (handle: Handle) => handle.fate === 'map' || handle.fate === 'copy';
+    for (const handle of recipe.handles.filter(isMapOrCopyFate)) {
```
Comment (csilvestrini) [inline src/tools/storage-key-recipe-resolver.ts:131]: It's a little strange defining the lambda and then using it in just the one place immediately below. I'd suggest this instead:

```
const mapOrCopyHandles = recipe.handles.filter(
    handle => handle.fate === 'map' || handle.fate === 'copy');
for (const handle of mapOrCopyHandles) { ... }
```

File: java/arcs/core/data/testdata/Request.arcs [hunk 5]
```diff
@@ -0,0 +1,29 @@
+// Big example for recipe2plan generation.
```
Comment (csilvestrini) [inline java/arcs/core/data/testdata/Request.arcs:1]: These files all seem way too complicated just for a test. The details of what they are supposed to be testing are completely lost in all the domain-specific names.

Can you please rewrite these to be much clearer in what they are designed to test? e.g. if you just want to test that an import works, maybe call the files something like `Importer.arcs` and `ManifestToImport.arcs` or something like that. And then simplify and rename the particles and schemas to just test the things you need to test, with simpler names.

File: java/arcs/core/data/testdata/BUILD (hunk 1 above)
Comment (alxrsngrtn) [reply java/arcs/core/data/testdata/BUILD:41]: Sorry -- Extract, Transform, Load.

File: javatests/arcs/android/e2e/ArcsTests.kt (hunk 2 above)
Comment (alxrsngrtn) [reply javatests/arcs/android/e2e/ArcsTests.kt:99]: I'll remove the ignore -- I added this last week, so it's certainly no longer relevant.

File: src/tools/recipe2plan.ts (hunk 3 above)
Comment (alxrsngrtn) [reply src/tools/recipe2plan.ts:25]: `parseFile` doesn't seem to import other manifests.

File: src/tools/storage-key-recipe-resolver.ts (hunk 4 above)
Comment (alxrsngrtn) [reply src/tools/storage-key-recipe-resolver.ts:131]: I'm happy with that. I was trying to get the whitespace right.

File: java/arcs/core/data/testdata/Request.arcs (hunk 5 above)
Comment (alxrsngrtn) [reply java/arcs/core/data/testdata/Request.arcs:1]: The intention of these manifests is to be a complex, realistic example. They are a broken up version of [this manifest](https://github.com/PolymerLabs/arcs/blob/master/java/arcs/core/data/testdata/IngestProcessRetrieve.arcs).

Comment (csilvestrini) [reply java/arcs/core/data/testdata/Request.arcs:1]: I'd recommend reworking this to be smaller, simpler unit tests instead, that specifically address the test cases you're interested in.

The complex, realistic example will be the actual client usages, which will fail at compile time if there's a problem. We should have a thorough battery of unit tests to check all of the individual edge cases we expect to work, which hopefully means the complex read-world cases don't break.

It's really hard to know what's going on with these tests. If I broke one, I wouldn't know why, or how to fix it. Tests should be simple and readable.

File: java/arcs/core/data/testdata/BUILD (hunk 1 above)
Comment (csilvestrini) [reply java/arcs/core/data/testdata/BUILD:41]: Can you rename this then, that wasn't readily apparent.

File: src/tools/recipe2plan.ts (hunk 3 above)
Comment (csilvestrini) [reply src/tools/recipe2plan.ts:25]: That's surprising. Do you know why? Do you think we can fix it?

Comment (alxrsngrtn) [reply src/tools/recipe2plan.ts:25]: Testing things more -- you're right! This isn't the fix, I needed to call `Runtime.init(...)`.

File: java/arcs/core/data/testdata/Request.arcs (hunk 5 above)
Comment (alxrsngrtn) [reply java/arcs/core/data/testdata/Request.arcs:1]: I have made these examples simpler. You're right, these should be understandable to the maintainer. We do have unit tests for the rest of the system. These will help users know *why* things are going wrong, in specific. I'll expand on these unit tests to also exercise this feature -- and at a finer granularity.

File: src/tools/tests/goldens/WriterReaderExample.kt [hunk 6]
```diff
@@ -17,9 +17,7 @@ object IngestionPlan : Plan(
             "arcs.core.data.testdata.Reader",
             mapOf(
                 "data" to HandleConnection(
-                    StorageKeyParser.parse(
-                        "reference-mode://{db://25e71af4e9fc8b6958fc46a8f4b7cdf6b5f31516@arcs/Thing}{db://25e71af4e9fc8b6958fc46a8f4b7cdf6b5f31516@arcs/!:writingArcId/handle/my-handle-id}"
-                    ),
+                    CreateableStorageKey("my-handle-id", Capabilities.Persistent),
```
Comment (alxrsngrtn) [inline src/tools/tests/goldens/WriterReaderExample.kt:20]: @mmandlis Does this look right?

File: java/arcs/core/data/testdata/ConsumeProduce.arcs [hunk 7]
```diff
@@ -0,0 +1,23 @@
+meta
+  namespace: arcs.core.data.testdata.gen
+
+import './Schemas.arcs'
+import './Produce.arcs'
```
Comment (mmandlis) [inline java/arcs/core/data/testdata/ConsumeProduce.arcs:5]: why do you need to import Produce.arcs?

File: java/arcs/core/data/testdata/Consume.arcs [hunk 8]
```diff
@@ -0,0 +1,18 @@
+// Big example for recipe2plan generation.
+
+meta
+  namespace: arcs.core.data.testdata.gen
+
+import './Schemas.arcs'
+import './ConsumeProduce.arcs'
```
Comment (mmandlis) [inline java/arcs/core/data/testdata/Consume.arcs:7]: why do you need to import ConsumeProduce.arcs?

File: src/tools/tests/goldens/WriterReaderExample.kt (hunk 6 above)
Comment (mmandlis) [reply src/tools/tests/goldens/WriterReaderExample.kt:20]: no, IngestionPlan is a long running arc (aka singleton recipe).
its storage keys need to be generated at compile time.

this applies to all changes in this file.

Comment (alxrsngrtn) [reply src/tools/tests/goldens/WriterReaderExample.kt]: Thanks for checking, I've found / implemented a fix.

File: java/arcs/core/data/testdata/ConsumeProduce.arcs (hunk 7 above)
Comment (piotrswigon) [reply java/arcs/core/data/testdata/ConsumeProduce.arcs:5]: recipe2plan operates in terms of a given manifest, if you don't import the .arcs file from which you map it will not be able to find a store.

... but. What we could do is to refactor things so that you can pass multiple files to recipe2plan and then construct a manifest that imports all the passed files and that would be your context?

Comment (mmandlis) [reply java/arcs/core/data/testdata/ConsumeProduce.arcs:5]: oh, i see. 
another option is to create a single top level arcs file that imports everything, like the "canonical" we used to have in the web arcs.
